File: docs/fortran/algorithmes/event_handler.f90
!  First programs in Fortran
!  Julien VALENTIN (2022)  julien.vlnt@gmail.com
!  Count until the user hits a key.
program Count
    
    use ifcore

    ! Initialize the counter
    integer :: var = 0

    ! Counting loop
    do while(.true.)
        
        print '(I0)', var

        ! Update
        var = var + 1

        ! If a key has been pressed : exit
        if(peekcharqq()) then
            print *, "Terminate after receiving signal : ", getcharqq()
            call abort()
        end if

        ! To count slow enough...
        call sleep(1)

        ! Watchdog
        if(var > 25) then
            exit
        end if

    end do

end program

File: docs/python3/algorithmes/event_handler.py
#  First programs in Python
#  Julien VALENTIN (July, 2022)  julien.vlnt@gmail.com
#  Count until the user hits a key.
from pynput import keyboard
from time import sleep

# Boolean `kill` initialized to false.
kill = False


# Function called when a key is pressed.
def on_press(key):
    "Turns a boolean variable to True when a key is pressed."
    
    # Change the scope
    global kill

    print("A key has been pressed.")
    kill = True

    return False


# Instance of the listener
listener = keyboard.Listener(on_press=on_press)

# Start the listener
listener.start()

# Declare the counter
count = 0

# Start to count...
while(not kill):

    print(count)

    # Update
    count += 1
    
    # To count slow enough...
    sleep(1)

    # Watchdog
    if(count > 100):
        break

# What does it mean ?
listener.join()
